services/src/main/java/org/apache/druid/server/router/TieredBrokerHostSelector.java

```diff
@@ -257,6 +268,42 @@ public String getDefaultServiceName()
     return new Pair<>(brokerServiceName, nodesHolder.pick());
   }
 
+  public Pair<String, Server> selectForSql(SqlQuery sqlQuery)
+  {
+    synchronized (lock) {
```

Comment:
Can part of this method be refactor to share common code with the select method?

Reply:
actually, I just realized that this block is not even required for `selectForSql` since `ruleManager` is not used to select the target server for sql query. @kfaraz - shall this be fixed?

Reply:
Thanks for catching this, @abhishekagarwal87 . We still need the block as we need to check on the `started` field of the `TieredBrokerHostSelector` but we don't need the other condition.

I am pushing the fix.